rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -42,57 +38,83 @@
+        self._load_model_instance(self._config["skip_model_load"])
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns LanguageModelFeaturizer's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # name of the language model to load.
+            "model_name": "bert",
+            # Pre-Trained weights to be loaded(string)
+            "model_weights": None,
+            # an optional path to a specific directory to download
+            # and cache the pre-trained model weights.
+            "cache_dir": None,
+            "skip_model_load": False,
+            "language": None,
+        }
 
     @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        """Packages needed to be installed."""
-        return [Tokenizer]
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
+        """Validates the configuration."""
+        pass
 
-    def __init__(
-        self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        skip_model_load: bool = False,
+    @classmethod
+    def validate_compatibility_with_tokenizer(
+        cls, config: Dict[Text, Any], tokenizer_type: Type[Tokenizer]
     ) -> None:
-        """Initializes LanguageModelFeaturizer with the specified model.
-
-        Args:
-            component_config: Configuration for the component.
-            skip_model_load: Skip loading the model for pytests.
-        """
-        super(LanguageModelFeaturizer, self).__init__(component_config)
-        self._load_model_metadata()
-        self._load_model_instance(skip_model_load)
+        """Checks that the featurizer and tokenizer are compatible."""
+        pass
 
     @classmethod
     def create(
-        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
-    ) -> "DenseFeaturizer":
-        language = config.language
-        if not cls.can_handle_language(language):
-            # check failed
-            raise UnsupportedLanguageError(cls.name, language)
-        return cls(component_config)
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "LanguageModelFeaturizerGraphComponent":
+        """Creates a LanguageModelFeaturizer.
+
+        Loads the model specified in the config.
+        """
+        return cls(config, execution_context)
+
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Returns the extra python dependencies required."""
+        return ["transformers"]
+
+    def _check_language_is_supported(self) -> None:
+        if "language" in self._config:
+            language = self._config["language"]
+            supported_languages = self.supported_languages()
+            if supported_languages is not None and language not in supported_languages:
+                raise UnsupportedLanguageError(self.__name__, language)
```

Comment:
@wochinge the language parameter and checks can be removed here as well, right? -- btw, what do we do when the supported language effectively depends on the config (e.g. because a model trained for spanish is loaded... (I have no idea if they can differ here but seems reasonable that they maybe will differ))?

Reply:
Good point 👍🏻 I think the best would probably be to add a config param to `supported_languages`? (cc @joejuzl ). Then the `supported_languages` can decide based on its config.

Reply:
@aeshky We can dump the entire `_check_language_is_supported` check for now as we do this as part of the [graph schema validation](https://github.com/RasaHQ/rasa/blob/262b5d1dcae599b56e21a81e421f7462699b9a1c/rasa/engine/validation.py#L133). 

Makes sense to add the extended check (having a dynamic `supported_languages` using the config) in the future but for now this is in my opinion not needed as we haven't done it before and we focus on the migration now

Reply:
>Makes sense to add the extended check (having a dynamic supported_languages using the config) in the future

I added a `TODO` for this.

Reply:
Can we please rather create an issue @aeshky ? Todos in the code tend to never get done and just make the code messy